### PR TITLE
Allow ECR image retrieval for the lambda role

### PR DESCRIPTION
**Pull request by gfou-al**

### CODE REVIEW

File: terraform/environments/core-shared-services/iam.tf [hunk 1]
```diff
@@ -162,7 +162,28 @@ data "aws_iam_policy_document" "instance-scheduler-lambda-function-policy" {
...
     resources = [
       format("arn:aws:secretsmanager:eu-west-2:%s:secret:environment_management*", local.environment_management.modernisation_platform_account_id)
-   ]
+    ]
+  }
+  statement {
+    sid    = "AllowECRImageRetrieval"
+    effect = "Allow"
+    actions = [
+      "ecr:SetRepositoryPolicy",
```
Comment (julialawrence) [inline terraform/environments/core-shared-services/iam.tf:171]: Is this permission needed for pulling images?

File: terraform/environments/core-shared-services/iam.tf [hunk 2]
```diff
@@ -162,7 +162,28 @@ data "aws_iam_policy_document" "instance-scheduler-lambda-function-policy" {
...
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage",
+      "ecr:CompleteLayerUpload",
```
Comment (julialawrence) [inline terraform/environments/core-shared-services/iam.tf:174]: Same here. Does lambda need upload permissions?

File: terraform/environments/core-shared-services/iam.tf [hunk 3]
```diff
@@ -162,7 +162,28 @@ data "aws_iam_policy_document" "instance-scheduler-lambda-function-policy" {
...
+      "ecr:DescribeImages",
+      "ecr:DescribeRepositories",
+      "ecr:UploadLayerPart",
```
Comment (julialawrence) [inline terraform/environments/core-shared-services/iam.tf:177]: Same here, upload permissions. If these are needed that's fine, but are theu#

File: terraform/environments/core-shared-services/iam.tf [hunk 4]
```diff
@@ -162,7 +162,28 @@ data "aws_iam_policy_document" "instance-scheduler-lambda-function-policy" {
...
+      "ecr:ListImages",
+      "ecr:InitiateLayerUpload",
```
Comment (julialawrence) [inline terraform/environments/core-shared-services/iam.tf:179]: Again, uploading. Is lambda likely to do this
?

File: terraform/environments/core-shared-services/iam.tf [hunk 5]
```diff
@@ -162,7 +162,28 @@ data "aws_iam_policy_document" "instance-scheduler-lambda-function-policy" {
...
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetRepositoryPolicy",
+      "ecr:PutImage"
```
Comment (julialawrence) [inline terraform/environments/core-shared-services/iam.tf:182]: We are looking to get image not put it I don't think.

**Review by julialawrence — COMMENTED** [review]
Just some niggles with the perms.

**Comment by gfou-al** [comment]
Closing this PR because the additional permissions were not required. The original error mislead us to believe we had to add permissions, but after locally re-applying through terraform the lambda a few times, the error was gone. It could have been related to the fact that the lambda deployment caches out the image instead of pulling the latest image.

**Pull request closed without merging**